ast/compile.go

```diff
@@ -2758,7 +2774,10 @@ func expandExprTermSlice(gen *localVarGenerator, v []*Term) (support []*Expr) {
 	return
 }
 
-type localDeclaredVars []*declaredVarSet
+type localDeclaredVars struct {
+	vars      []*declaredVarSet
+	rewritten map[Var]Var
```

Comment:
It might be useful to add a comment that explains why this exists on the struct (since it's somewhat redundant with the stack of var sets above.) Something along the lines of "rewritten contains a mapping of user-defined variables that have been rewritten whereas vars only contains rewrite state from the current query (not not any nested queries.)"

Reply:
Will do

Reply:
Done.